tests/unit/setup-rendering-context-test.js

```diff
@@ -0,0 +1,140 @@
+import { module, test } from 'qunit';
+import Service from '@ember/service';
+import Component from '@ember/component';
+import { helper } from '@ember/component/helper';
+import {
+  setupContext,
+  setupRenderingContext,
+  teardownContext,
+} from 'ember-test-helpers';
+import hasEmberVersion from 'ember-test-helpers/has-ember-version';
+import hasjQuery from '../helpers/has-jquery';
+import { setResolverRegistry } from '../helpers/resolver';
+import hbs from 'htmlbars-inline-precompile';
+
+module('setupRenderingContext', function(hooks) {
+  if (!hasEmberVersion(2, 4)) {
+    return;
+  }
+
+  hooks.before(function() {
+    setResolverRegistry({
+      'service:foo': Service.extend({ isFoo: true }),
+      'template:components/template-only': hbs`template-only component here`,
+      'component:js-only': Component.extend({
+        classNames: ['js-only'],
+      }),
+      'helper:jax': helper(([name]) => `${name}-jax`),
+      'template:components/outer-comp': hbs`outer{{inner-comp}}outer`,
+      'template:components/inner-comp': hbs`inner`,
+    });
+  });
+
+  hooks.beforeEach(function() {
+    setupContext(this);
+    setupRenderingContext(this);
+  });
+
+  hooks.afterEach(function() {
+    teardownContext(this);
+  });
+
+  test('render exposes an `.element` property', function(assert) {
+    return this.render(hbs`<p>Hello!</p>`).then(() => {
+      assert.equal(this.element.textContent, 'Hello!');
+    });
+  });
+
+  test('render can be used multiple times', function(assert) {
+    return this.render(hbs`<p>Hello!</p>`)
+      .then(() => {
+        assert.equal(this.element.textContent, 'Hello!');
+
+        return this.render(hbs`<p>World!</p>`);
+      })
+      .then(() => {
+        assert.equal(this.element.textContent, 'World!');
+      });
+  });
+
+  test('render does not run sync', function(assert) {
+    assert.equal(
+      this.element,
+      undefined,
+      'precond - this.element is not set before this.render'
+    );
+
+    let renderPromise = this.render(hbs`<p>Hello!</p>`).then(() => {
+      assert.equal(this.element.textContent, 'Hello!');
+    });
+
+    assert.equal(
+      this.element,
+      undefined,
+      'precond - this.element is not set sync after this.render'
+    );
+
+    return renderPromise.then(() => {
+      assert.equal(this.element.textContent, 'Hello!');
+    });
```

Comment:
This seems redundant since the `.then()` above has already checked the same thing?!